pkg/kubelet/kubelet.go

```diff
@@ -2244,6 +2244,27 @@ func (kl *Kubelet) getPodsToSync() []*v1.Pod {
 	return podsToSync
 }
 
+// Ensure node is registered before syncing static pods.
+func (kl *Kubelet) syncStaticPodsOnNodeRegistration() {
+	nodeCh := make(chan struct{})
+	wait.Until(func() {
+		if _, err := kl.GetNode(); err != nil {
+			return
+		}
+		close(nodeCh)
+	}, time.Second, nodeCh)
```

Comment:
@AnishShah This will block Kubelet until GetNode succeeds. Is it ok from your point of view?